added experimentation metric CRUD module

cosmic-flood opened the pull request:
<!--
# Instructions

1. Pick a meaningful title for your pull request. (Use sentence case. Keep it short < 70 characters)
  - Prefix the title with an emoji to categorize what is being done. (Copy-paste the emoji from the list below.)
  - Assign the appropriate label(s) to the pull request. (Use one of the labels from the list below.)
  
2. Enter a succinct description that says why the PR is necessary, and what it does.
  - Implement aspect X
  - Leave out feature Y because of A
  - Improve performance by B
  - Improve accessibility by C

3. Provide clear testing instructions that include any pertinent information, i.e. seat, roles, etc.
4. Include screenshots of your changes if they impact the UI (Before & After).
5. Update the preview link with your pull request number.
6. Include any JIRA tickets, design documents, GitHub issues, or other related items to the bottom of the PR.

# Available labels
UI
API
Evaluation Server
OLAP

# Emojis for categorizing pull requests

✨ New feature or functionality
🐛 Bugfix
🔥 P0 fix
✅ Tests
🚀 Performance improvements
🖍 CSS / Styling
📖 Documentation
🏗 Infrastructure / Tooling / Builds / CI
↩️ Reverting a previous change
🧹 Refactoring / Housekeeping
🔧 Package Upgrades / Maintenance
🗑️ Deleting code

-->

deleteLater commented on modules/front-end/src/app/features/safe/experiments/metrics/metrics.component.ts:
```diff
@@ -45,14 +45,14 @@ export class MetricsComponent implements OnInit, OnDestroy {
+    //         this.onCreateOrEditClick(res);
+    //       }
+    //     });
+    //   }
```
why keep these unused code

cosmic-flood replied:
Not my code...

cosmic-flood replied:
not my code~  keep it

deleteLater commented on modules/back-end/src/Api/Controllers/ExperimentMetricController.cs:
```diff
@@ -0,0 +1,55 @@
+    }
+
+    [HttpDelete("{id:guid}")]
+    public async Task<ApiResponse<bool>> ArchiveAsync(Guid id)
```
just use delete

deleteLater commented on modules/front-end/src/app/core/components/metric-drawer/metric-drawer.component.html:
```diff
@@ -51,8 +51,8 @@ <h3 i18n="@@expt.metric.event-info">Event information</h3>
             <nz-option [nzValue]="customEventType"  nzLabel="Custom"></nz-option>
-            <nz-option [nzValue]="pageViewEventType" nzLabel="PageView"></nz-option>
-            <nz-option [nzValue]="clickEventType" nzLabel="Click"></nz-option>
+            <!-- <nz-option [nzValue]="pageViewEventType" nzLabel="PageView"></nz-option>
+            <nz-option [nzValue]="clickEventType" nzLabel="Click"></nz-option> -->
```
why comment

cosmic-flood replied:
We need to leave this part for now. we will uncomment when another part is ready. We now support only custom event for an easy mode.

cosmos-explorer replied:
why leave for now?

deleteLater commented on modules/back-end/src/Infrastructure/Experiments/ExperimentMetricService.cs:
```diff
@@ -0,0 +1,53 @@
+
+        public async Task DeleteAsync(Guid id)
+        {
+            await MongoDb.CollectionOf<ExperimentMetric>().DeleteOneAsync(x => x.Id == id);
```
just use `Collection.DeleteOneAsync(...)` is ok, see base class

deleteLater commented on modules/back-end/src/Domain/Experiments/ExperimentMetric.cs:
```diff
@@ -0,0 +1,19 @@
+    public string ElementTargets { get; set; }
+    public List<TargetUrl> TargetUrls { get; set; }
+
+    public bool IsArvhived { get; set; }
```
typo and why this field

cosmos-explorer replied:
Agree

cosmic-flood replied:
This is the original prodduct requirement, we didn't officially change it yet. We keep Archive instead of Delete

cosmic-flood commented:
@cosmos-explorer   can you review my PR and make your comment? If nothing to correct, can you approve my PR?

cosmos-explorer commented on modules/front-end/src/app/core/services/metric.service.ts:
```diff
@@ -3,17 +3,24 @@ import { Injectable } from '@angular/core';
+  get baseUrl() {
+    return environment.url + `/api/v1/envs/${this.envId}/experiment-metrics`;
+  }
+  // baseUrl: string = environment.url + `/api/v1/envs/${this.envId}/experiment-metrics`;
```
remove the comment

cosmic-flood commented:
![image](https://user-images.githubusercontent.com/68597908/197378155-7e7866b6-aa97-4a65-9b1f-a5ed604da89b.png)

Updated information:

front-end：
1. Removed useless commented code
2. Added a i18n key for quick start module.
3. Added a link of "Go to document" in quick start guide drawer.

backend：
1. Resolve a conflict with current main. Added "IExperimentMetricService" in [ConfigureServices.cs]

@cosmos-explorer @deleteLater

cosmos-explorer commented on modules/front-end/src/app/core/components/menu/menu.component.html:
```diff
@@ -2,7 +2,9 @@
     <div class="brand-content">
       <a class="brand-logo" [routerLink]="['/']">
-        <img src="assets/icons/logo-with-text.png" alt=""> 
+        <!-- <img src="assets/logo/logo-with-text.png" alt="">  -->
```
why always keep the old code?

cosmos-explorer commented on modules/front-end/src/app/core/components/menu/menu.component.less:
```diff
@@ -69,17 +69,35 @@ section.menu-body {
+        //   width: 149.07px;
+        //   height: 58px;
+        //   object-fit: cover;
+        // }
```
same

cosmos-explorer commented on modules/front-end/src/app/features/login/login.component.html:
```diff
@@ -4,8 +4,9 @@
-        <div class="brand">FeatBit</div>
+        <img class="logo" src="assets/logo/logo_black_transparent.svg">
+        <img class="brand" src="assets/logo/logo_title_black.svg">
+        <!-- <div class="brand">FeatBit</div> -->
```
the same
